DOStateStore `__dirname is not defined`

I get the error below while using DOStateStore on Alchemy v0.37.1. I believe this is an ESM error and the culprit is here https://github.com/sam-goodwin/alchemy/blob/73b907b52ff6c9ed2528fec1b557eb9e4cb2f6f6/alchemy/src/cloudflare/do-state-store/internal.ts#L205. Perhaps because I am using pnpm?

Code: 
```
const infra = await alchemy(APP_NAME, {
  stage: stage,
  phase: phase,
  quiet,
  password: process.env.SECRET_ALCHEMY_PASSPHRASE,
  stateStore: (scope) =>
    new DOStateStore(scope, {
      // apiKey: alchemy.secret(process.env.CLOUDFLARE_API_KEY),
      // email: process.env.CLOUDFLARE_EMAIL,
      worker: {
        name: `${APP_NAME}-${stage}-alchemy-state`,
      },
    }),
})
```

Error:
```
Alchemy (v0.37.1)
App: solstatus
Phase: up
Stage: prod
ERROR Scope failed solstatus/prod
node:internal/modules/run_main:129
    triggerUncaughtException(
    ^

ReferenceError: __dirname is not defined
    at bundleWorkerScript (/Users/jbeckman/projects/personal/solstatus-personal/node_modules/.pnpm/alchemy@0.37.1_@ai-sdk+openai-compatible@0.2.14_zod@3.25.67__@ai-sdk+openai@1.3.22_zod@_3a6d83a9a73bf53db983440fc3b33974/node_modules/alchemy/src/cloudflare/do-state-store/internal.ts:221:27)
    at upsertStateStoreWorker (/Users/jbeckman/projects/personal/solstatus-personal/node_modules/.pnpm/alchemy@0.37.1_@ai-sdk+openai-compatible@0.2.14_zod@3.25.67__@ai-sdk+openai@1.3.22_zod@_3a6d83a9a73bf53db983440fc3b33974/node_modules/alchemy/src/cloudflare/do-state-store/internal.ts:120:38)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Promise.all (index 1)
    at DOStateStore.createClient (/Users/jbeckman/projects/personal/solstatus-personal/node_modules/.pnpm/alchemy@0.37.1_@ai-sdk+openai-compatible@0.2.14_zod@3.25.67__@ai-sdk+openai@1.3.22_zod@_3a6d83a9a73bf53db983440fc3b33974/node_modules/alchemy/src/cloudflare/do-state-store/store.ts:89:28)
    at DOStateStore.getClient (/Users/jbeckman/projects/personal/solstatus-personal/node_modules/.pnpm/alchemy@0.37.1_@ai-sdk+openai-compatible@0.2.14_zod@3.25.67__@ai-sdk+openai@1.3.22_zod@_3a6d83a9a73bf53db983440fc3b33974/node_modules/alchemy/src/cloudflare/do-state-store/store.ts:129:12)
    at DOStateStore.init (/Users/jbeckman/projects/personal/solstatus-personal/node_modules/.pnpm/alchemy@0.37.1_@ai-sdk+openai-compatible@0.2.14_zod@3.25.67__@ai-sdk+openai@1.3.22_zod@_3a6d83a9a73bf53db983440fc3b33974/node_modules/alchemy/src/cloudflare/do-state-store/store.ts:133:5)
    at async Promise.all (index 0)
    at Scope.init (/Users/jbeckman/projects/personal/solstatus-personal/node_modules/.pnpm/alchemy@0.37.1_@ai-sdk+openai-compatible@0.2.14_zod@3.25.67__@ai-sdk+openai@1.3.22_zod@_3a6d83a9a73bf53db983440fc3b33974/node_modules/alchemy/src/scope.ts:170:5)
    at _apply (/Users/jbeckman/projects/personal/solstatus-personal/node_modules/.pnpm/alchemy@0.37.1_@ai-sdk+openai-compatible@0.2.14_zod@3.25.67__@ai-sdk+openai@1.3.22_zod@_3a6d83a9a73bf53db983440fc3b33974/node_modules/alchemy/src/apply.ts:45:5)

Node.js v20.13.1

❌ Command failed: Error: Command failed: tsx /Users/jbeckman/projects/personal/solstatus-personal/packages/infra/src/alchemy.run.ts up prod
[11:30:39.927] ERROR (#17):
  Error: 1
 ELIFECYCLE  Command failed with exit code 1.
```

```
✘ [ERROR] Could not resolve "/Users/jbeckman/projects/personal/solstatus/node_modules/.pnpm/alchemy@https+++pkg.pr.new+sam-goodwin+alchemy@452_@ai-sdk+openai-compatible@0.2.14_zod_33ac7b35f29ac8db5b0e199d8c595d8e/node_modules/alchemy/lib/cloudflare/do-state-store/worker.ts"
```

> ```
> ✘ [ERROR] Could not resolve "/Users/jbeckman/projects/personal/solstatus/node_modules/.pnpm/alchemy@https+++pkg.pr.new+sam-goodwin+alchemy@452_@ai-sdk+openai-compatible@0.2.14_zod_33ac7b35f29ac8db5b0e199d8c595d8e/node_modules/alchemy/lib/cloudflare/do-state-store/worker.ts"
> ```

Ok that problem is caused because it's looking for `.ts` which can only work if you run with `bun`. Fixing.

--- a/alchemy/src/cloudflare/do-state-store/internal.ts
+++ b/alchemy/src/cloudflare/do-state-store/internal.ts
@@ -1,6 +1,7 @@
 import path from "node:path";
 import { fileURLToPath } from "node:url";
 import { bundle } from "../../esbuild/index.ts";
+import { exists } from "../../util/exists.ts";
 import { withExponentialBackoff } from "../../util/retry.ts";
 import { handleApiError } from "../api-error.ts";
 import type { CloudflareApi } from "../api.ts";
@@ -218,13 +219,15 @@ export async function getAccountSubdomain(api: CloudflareApi) {
 }
 
 async function bundleWorkerScript() {
+  const thisDir =
+    typeof __dirname !== "undefined"
+      ? __dirname
+      : path.dirname(fileURLToPath(import.meta.url));
+  // we can be running from .ts or .js, so resolve it defensively
+  const workerTs = path.join(thisDir, "worker.ts");
+  const workerJs = path.join(thisDir, "worker.js");
   const result = await bundle({
-    entryPoint: path.join(
-      typeof __dirname !== "undefined"
-        ? __dirname
-        : path.dirname(fileURLToPath(import.meta.url)),
-      "worker.ts",
-    ),
+    entryPoint: (await exists(workerTs)) ? workerTs : workerJs,
     bundle: true,
     format: "esm",
     target: "es2022",

--- a/alchemy/src/cloudflare/do-state-store/internal.ts
+++ b/alchemy/src/cloudflare/do-state-store/internal.ts
@@ -7,28 +7,6 @@ import type { CloudflareApi } from "../api.ts";
 import type { WorkerMetadata } from "../worker-metadata.ts";
 import type { DOStateStoreAPI } from "./types.ts";
 
-/**
- * Get the current directory in a way that works in both ESM and CommonJS environments
- */
-function getCurrentDir(): string {
-  // First try import.meta.dir (Bun)
-  if (typeof import.meta !== 'undefined' && 'dir' in import.meta) {
-    return import.meta.dir as string;
-  }
-  
-  // Then try import.meta.url (Node.js ESM)
-  if (typeof import.meta !== 'undefined' && import.meta.url) {
-    return path.dirname(fileURLToPath(import.meta.url));
-  }
-  
-  // Fallback for CommonJS (though this shouldn't be needed in this ESM context)
-  if (typeof __dirname !== 'undefined') {
-    return __dirname;
-  }
-  
-  throw new Error('Unable to determine current directory - unsupported environment');
-}
-
 interface DOStateStoreClientOptions {
   app: string;
   stage: string;
@@ -241,7 +219,12 @@ export async function getAccountSubdomain(api: CloudflareApi) {
 
 async function bundleWorkerScript() {
   const result = await bundle({
-    entryPoint: path.join(getCurrentDir(), "worker.ts"),
+    entryPoint: path.join(
+      typeof __dirname !== "undefined"
+        ? __dirname
+        : path.dirname(fileURLToPath(import.meta.url)),
+      "worker.ts",
+    ),
     bundle: true,
     format: "esm",
     target: "es2022",

--- a/alchemy/src/cloudflare/do-state-store/internal.ts
+++ b/alchemy/src/cloudflare/do-state-store/internal.ts
@@ -1,10 +1,33 @@
 import path from "node:path";
+import { fileURLToPath } from "node:url";
 import { bundle } from "../../esbuild/index.ts";
 import { withExponentialBackoff } from "../../util/retry.ts";
 import { handleApiError } from "../api-error.ts";
 import type { CloudflareApi } from "../api.ts";
 import type { WorkerMetadata } from "../worker-metadata.ts";
 import type { DOStateStoreAPI } from "./types.ts";
+
+/**
+ * Get the current directory in a way that works in both ESM and CommonJS environments
+ */
+function getCurrentDir(): string {
+  // First try import.meta.dir (Bun)
+  if (typeof import.meta !== 'undefined' && 'dir' in import.meta) {
+    return import.meta.dir as string;
+  }
+  
+  // Then try import.meta.url (Node.js ESM)
+  if (typeof import.meta !== 'undefined' && import.meta.url) {
+    return path.dirname(fileURLToPath(import.meta.url));
+  }
+  
+  // Fallback for CommonJS (though this shouldn't be needed in this ESM context)
+  if (typeof __dirname !== 'undefined') {
+    return __dirname;
+  }
+  
+  throw new Error('Unable to determine current directory - unsupported environment');
+}
 
 interface DOStateStoreClientOptions {
   app: string;
@@ -218,7 +241,7 @@ export async function getAccountSubdomain(api: CloudflareApi) {
 
 async function bundleWorkerScript() {
   const result = await bundle({
-    entryPoint: path.join(__dirname, "worker.ts"),
+    entryPoint: path.join(getCurrentDir(), "worker.ts"),
     bundle: true,
     format: "esm",
     target: "es2022",

--- a/alchemy/src/util/exists.ts
+++ b/alchemy/src/util/exists.ts
@@ -1,0 +1,10 @@
+import { access } from "node:fs/promises";
+
+export async function exists(path: string): Promise<boolean> {
+  try {
+    await access(path);
+    return true;
+  } catch {
+    return false;
+  }
+}

